src/pages/settings/SettingForm.tsx

```diff
@@ -172,6 +170,10 @@ const SettingForm: FC<Props> = ({
                   initialValue={value ?? ""}
                   configField={configField}
                   onSubmit={onSubmit}
+                  onDelete={(key: string) => {
+                    onDelete(key);
+                    setEditMode(false);
+                  }}
```

Comment:
```suggestion
                  onDelete={onDelete}
```
A bit simpler, the field vanishes so we don't need to disable edit mode. On failure, it is preferred to stay in edit mode, so the user can retry.